app/src/main/java/com/alphawallet/app/walletconnect/WalletConnectV2SessionRequestHandler.java

```diff
@@ -61,19 +68,74 @@ private void showDialog(String method, ActionSheetCallback aCallback)
         BaseRequest signRequest = EthSignRequest.getSignRequest(sessionRequest);
         if (signRequest != null)
         {
-            Signable signable = signRequest.getSignable(sessionRequest.getRequest().getId(), settledSession.getMetaData().getUrl());
-            ActionSheet actionSheet = new ActionSheetSignDialog(activity, aCallback, signable);
-            actionSheet.setSigningWallet(signRequest.getWalletAddress());
-            List<String> icons = Objects.requireNonNull(settledSession.getMetaData()).getIcons();
-            if (!icons.isEmpty())
+            Signable signable = signRequest.getSignable(sessionRequest.getRequest().getId(), Objects.requireNonNull(settledSession.getMetaData()).getUrl());
+            WalletConnectV2SessionItem session = new WalletConnectV2SessionItem(settledSession);
+
+            List<Long> chainList = new ArrayList<>();
+            for (String chain : session.chains)
```

Comment:
Refactor to make this a bit more maintainable. Maybe move the chainList formation into a separate method.